### PR TITLE
Payout completed contract

### PR DESCRIPTION
**Pull request by cpacia**

### CODE REVIEW

File: api/restapi.py [hunk 1]
```diff
@@ -306,11 +306,11 @@ def update_profile(self, request):
             if "short_description" in request.args:
                 u.short_description = request.args["short_description"][0]
             if "nsfw" in request.args:
-                u.nsfw = True
+                u.nsfw = request.args["nsfw"][0]
             if "vendor" in request.args:
-                u.vendor = True
+                u.vendor = request.ags["vendor"][0]
```
Comment (gubatron) [inline api/restapi.py]: typo -> `s/ags/args`

File: api/restapi.py [hunk 2]
```diff
@@ -589,11 +593,24 @@ def respond(success):
...
-                hash_value = digest(image).encode("hex")
-                with open(DATA_FOLDER + "store/media/" + hash_value, 'w') as outfile:
-                    outfile.write(image)
-                self.db.HashMap().insert(digest(image), DATA_FOLDER + "store/media/" + hash_value)
+            if "image" in request.args:
+                for image in request.args["image"]:
+                    hash_value = digest(image).encode("hex")
+                    with open(DATA_FOLDER + "store/media/" + hash_value, 'w') as outfile:
+                        outfile.write(image)
+                    self.db.HashMap().insert(digest(image), DATA_FOLDER + "store/media/" + hash_value)
```
Comment (gubatron) [inline api/restapi.py]: perhaps it's better if the `digest(image)` is calculated only once and reused.

File: market/contracts.py [hunk 3]
```diff
@@ -420,6 +430,112 @@ def accept_order_confirmation(self, ws, confirmation_json=None):
...
+            receipt_json["buyer_receipt"]["receipt"]["rating"]["quality"] = quality
+            receipt_json["buyer_receipt"]["receipt"]["rating"]["description"] = description
+            receipt_json["buyer_receipt"]["receipt"]["rating"]["delivery_time"] = delivery_time
+            receipt_json["buyer_receipt"]["receipt"]["rating"]["customer_service"] = customer_service
+            receipt_json["buyer_receipt"]["receipt"]["rating"]["review"] = review
+        if payout:
+            vendor_guid = self.contract["vendor_offer"]["listing"]["id"]["guid"]
+            contract_id = self.contract["buyer_order"]["order"]["ref_hash"]
+            order_id = self.contract["vendor_order_confirmation"]["invoice"]["ref_hash"]
+            outpoints = pickle.loads(self.db.Purchases().get_outpoint(order_id))
```
Comment (gubatron) [inline market/contracts.py]: can we stay away from python specific serialization formats? not good for compatibility with other possible clients and tools.

Comment (gubatron) [inline market/contracts.py]: also a little out of order as I see the use of JSON and protobuf everywhere else.

File: market/contracts.py [hunk 4]
```diff
@@ -3,6 +3,7 @@
 import json
 import random
 import time
+import pickle
```
Comment (gubatron) [inline market/contracts.py]: :-1:

File: constants.py [hunk 5]
```diff
@@ -7,3 +7,4 @@
 SEED_NODE_TESTNET = ("205.186.154.163", 28467)
 KSIZE = 20
 ALPHA = 3
+TRANSACTION_FEE = 10000
```
Comment (gubatron) [inline constants.py]: is this satoshis? millibits? bitcoins? ;)

It'd be nice to have the unit in the name of the "constant"

File: api/wsclient.py [hunk 6]
```diff
@@ -38,7 +38,7 @@ def onMessage(self, payload, isBinary):
 
 if __name__ == '__main__':
 
-    factory = WebSocketClientFactory("ws://localhost:18466")
+    factory = WebSocketClientFactory("ws://seed.openbazaar.org:18466")
```
Comment (gubatron) [inline api/wsclient.py]: will this URL:port somehow be configurable in the future?
(say to avoid firewalls)

File: market/contracts.py (hunk 3 above)
Comment (hoffmabc) [inline market/contracts.py]: Didn't you hear Oracle is moving past Java? :)

File: api/wsclient.py (hunk 6 above)
Comment (hoffmabc) [inline api/wsclient.py]: are you asking for connecting to a seed or hosting a server in general?

File: market/contracts.py (hunk 3 above)
Comment (cpacia) [inline market/contracts.py]: What's an alternative for saving a list in the sqlite db?

File: constants.py (hunk 5 above)
Comment (cpacia) [inline constants.py]: Anytime you work with bitcoin in code it's in satoshis

File: market/contracts.py (hunk 3 above)
Comment (gubatron) [inline market/contracts.py]: haven't heard anything, in fact I'm going to JavaONE in SF this month.

@cpacia: can't save in JSON? or protobuf? something standard that other clients (not necessarily in java, I'm thinking Node, Ruby, C/C++ can handle?)

**Pull request merged**